From a8fad73e3334151196acb28c4dcde37732c82542 Mon Sep 17 00:00:00 2001
Message-Id: <a8fad73e3334151196acb28c4dcde37732c82542.1648734429.git.geliang.tang@suse.com>
From: Florent Revest <revest@chromium.org>
Date: Fri, 23 Apr 2021 01:55:43 +0200
Subject: [PATCH] bpf: Remove unnecessary map checks for ARG_PTR_TO_CONST_STR
Git-commit: a8fad73e3334151196acb28c4dcde37732c82542
Patch-mainline: v5.13-rc1
References: bsn#127
Modified-by-SEL: No


reg->type is enforced by check_reg_type() and map should never be NULL
(it would already have been dereferenced anyway) so these checks are
unnecessary.

Reported-by: Alexei Starovoitov <ast@kernel.org>
Signed-off-by: Florent Revest <revest@chromium.org>
Signed-off-by: Alexei Starovoitov <ast@kernel.org>
Link: https://lore.kernel.org/bpf/20210422235543.4007694-3-revest@chromium.org
Signed-off-by: Geliang Tang <geliang.tang@suse.com>
---
 kernel/bpf/verifier.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/kernel/bpf/verifier.c b/kernel/bpf/verifier.c
index 59799a9b014a..2579f6fbb5c3 100644
--- a/kernel/bpf/verifier.c
+++ b/kernel/bpf/verifier.c
@@ -5075,8 +5075,7 @@ static int check_func_arg(struct bpf_verifier_env *env, u32 arg,
 		u64 map_addr;
 		char *str_ptr;
 
-		if (reg->type != PTR_TO_MAP_VALUE || !map ||
-		    !bpf_map_is_rdonly(map)) {
+		if (!bpf_map_is_rdonly(map)) {
 			verbose(env, "R%d does not point to a readonly map'\n", regno);
 			return -EACCES;
 		}
-- 
2.34.1

